README.md

```diff
@@ -244,6 +255,43 @@ locale further down the line.
     if ($text === 'someValue') { ... } // Strict type check prevents calling the __toString() method
 ```
 
+## Translations for Doctrine column types other than `string`
+
+In fact, the fields that are used to hold translated values need not use the `translatable` Doctrine column type declaration
+and can be of other type than `string`. 
+
+In this case, you need to use a union type for your field type declaration as in the following example.
+
+```php
+<?php
+
+use Doctrine\ORM\Mapping as ORM;
+use Webfactory\Bundle\PolyglotBundle\Annotation as Polyglot;
+use Webfactory\Bundle\PolyglotBundle\TranslatableInterface;
+
+// ...
+class Document
+{
+    // ... fields and collections omitted for brevity
+
+    /**
+     * @ORM\Column(type=...your type) 
+     * @Polyglot\Translatable
+     */
+    private TranslatableInterface|<other type> $text;
+
+    // ...
+}
+```
+
+With this declaration, the field can serve a dual use: Right before the ORM flushes data, the field value will be switched to the 
+value of the "primary" locale, so that the ORM can persist that data as usual. Similarly, after the ORM has loaded the entity,
+it will replace the field value with the translation value holder (an instance of `TranslatableInterface`) that you can use
+to obtain the translated values as well.
+
+Note that it is not necessary to do this in the translations class (`DocumentTranslation` in the above examples), since that 
+class represents the values of a single locale only and never contains `TranslatableInterface` instances.
```

Comment:
Maybe we should mention that users could create their own Doctrine types modelled after our TranslatableType if they need them?

Reply:
Not sure... that's only for a very specialized use case of wanting type safety _and_ having translated values that are not strings but need a specialized type?

Reply:
Why shouldn't you want type safety when having specialised types? If that's something the bundle supports, and it is now type-safe by default, I think people would excpect the specialised types to be type-safe, too.

If you were in a case where you need custom types, and you could choose between dealing with null values that never exist in reality and copy-and-pasting a simple Doctrine type, which one would you choose? I think the one you would choose should be the documented one.

Reply:
I just realised that it's not just the Doctrine type, but also the `Uninitlialized…Translatable`, which is referenced in the `PersistentTranslatable`, which also makes the assumption it's a string.

As it would require extra development effort on the bundle side the to enable type safety for arbitrary types, I suggest not to invest it and leave the README as is.